tmplib/Makefile

```diff
@@ -1,10 +1,10 @@
-CC = riscv64-unknown-linux-gnu-gcc
+CC = riscv$(BITS)-unknown-linux-gnu-gcc
 CFLAGS = -Wall -Werror -fPIC -I.
 
 ASM_SRCS = uaccess.S
 ASM_OBJS = $(patsubst %.S,%.o,$(ASM_SRCS))
 
-BUILD_DEFS = -D__riscv_xlen=64 -D__ASSEMBLY__ -include partial_linkage.h
+BUILD_DEFS = -D__riscv_xlen=$(BITS) -D__ASSEMBLY__ -include partial_linkage.h
```

Comment:
Could you try to remove `-D__riscv_xlen=$(BITS)` and see if it still works?
As far as I remember, __riscv_xlen is automatically set. Check this thread:
https://groups.google.com/a/groups.riscv.org/forum/#!topic/sw-dev/EvYRsA2gUjc

Reply:
This has been removed and works.